### ISSUE TITLE
Campaign subjet email empty

### ISSUE THREAD

**Issue by lucalbert** [mautic/mautic#7286]
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description
Hello,

Yesterday I updated a mautic instance to version 2.15.0.

Since then, emails sent from the start of a campaign have an empty subject.

By cons, when I use the test tool of the email (send an example), the subject is well filled.

Same thing when I test the SMTP server: I have the subject.

Emails are sent from a custom SMTP server.

Thank you

| Q   | A
| --- | ---
| Mautic version | 2.15.0
| PHP version | 7.1.26
| Browser | Google Chrome 72.0.3626.109 (Build officiel) (64 bits)

### Steps to reproduce
1. Create email and campaign
2. Trigger campaign
 
### Log errors

_No error log_

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

**Comment by pety-dc** [comment]
Hey @lucalbert 

I experienced the same issue.  To be honest, I was surprised how long ago this issue was reported by you and yet nobody else ran into it. 

I spent my day trying to debug what can be the cause of the phenomena. And I concluded that in our case the root of the missing subject was @kuzmany's recommender plugin. 

In `MauticRecommenderBundle/EventListener/EmailSubscriber.php` the `onEmailGenerate` method set the subject as:

```
$event->setSubject($this->recommenderTokenReplacer->getRecommenderGenerator()->replaceTagsFromContent($event->getSubject()));
```

For yet unknown reasons this results in a null value set as subject of the email


If you happen to use this recommender plugin too, that may be your answer too.
Otherwise you can check whether another plugin does something similar. I did not find such error in the core code of mautic.

**Comment by lucalbert** [comment]
Hello @pety-dc,

Yes it's sure ! I provided a fix on March 7th: kuzmany/mautic-recommender-bundle#3
My patch works in production since this date.

Your fix (kuzmany/mautic-recommender-bundle#9) was released 2 days ago !

Thanks

**Comment by RCheesley** [comment]
Closing this down as it was likely a third-party plugin causing the problem.
